src/jit/compiler.hpp

```diff
@@ -1210,16 +1210,9 @@ inline GenTree* Compiler::gtNewFieldRef(var_types typ, CORINFO_FIELD_HANDLE fldH
     /* 'GT_FIELD' nodes may later get transformed into 'GT_IND' */
 
     assert(GenTree::s_gtNodeSizes[GT_IND] <= GenTree::s_gtNodeSizes[GT_FIELD]);
-    GenTree* tree = new (this, GT_FIELD) GenTreeField(typ);
+    GenTree* tree = new (this, GT_FIELD) GenTreeField(typ, obj, fldHnd, offset);
```

Comment:
Looks like now you can move `GenTree* tree = new (this, GT_FIELD) GenTreeField(typ, obj, fldHnd, offset); ` out of `#ifdef`. Also please add `#else // SMALL_TREE_NODES`.

Reply:
Hmm, SMALL_TREE_NODES, that's something we should delete